### ISSUE TITLE
further work

### ISSUE THREAD

**Issue by electronique** [InteractiveArtUMBC/jasoncharney#2]
one more thing. In see this work I also feel I would be ok if you would like to do more independent work during the semester but in a similar vein as you are doing now. I have made assignments that we all can work on over the semester but if you would like to work with me we can hone some of the more specific issue that you are having trouble with in your projects at this point. 

Saying that, one thing that might be interesting to explore is pull in some remote data sources into your sketches. I have some APIs that might be over interest that I could share and get you going with using JSON if this is of interest. 

Tell me your thoughts.

**Comment by jasoncharney** [comment]
Yes I'd very much like that! Even just working with p5/JavaScript this past week, I realized how much is possible and would like to go further in learning JavaScript outside of p5 (been looking at libraries like three.js). I am interested in using APIs for data, but what I really want to learn is how to use websockets like you mentioned last Friday to allow multiple users to connect to a server and have their devices expand a generative audio-visual environment, as a way to spatialize sound and light using client devices – not just mirroring what's going on on a big screen, but meaningfully becoming satellite A/V channels.

I followed Daniel Shiffman's tutorial on using p5 and sockets.io [https://www.youtube.com/watch?v=i6eP1Lw4gZk](url), but want to go further. I'm still pretty new to JavaScript so kinda unclear on how to set up callbacks, automatic connections, broadcast events for synchronous execution, etc. Even just going through the sockets.io documentation has me still confused. So I'd like to work on that kind of thing in particular. Do you have any recommendations for getting more into websockets?
